src/cache/cache_las.c

```diff
@@ -810,9 +808,28 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
 err:
     /* Resolve the transaction. */
     if (local_txn) {
-        if (ret == 0)
+        if (ret == 0) {
             ret = __wt_txn_commit(session, NULL);
-        else
+            if (ret == 0)
+                /* Traverse the keys again to flag the updates inserted to lookaside. */
```

Comment:
This change lets the loop traverse twice in a success scenario. I prefer set the flag whenever the insert takes place and in case of any error clear the flag. So only in failure scenarios, the loop traverse occurs twice and successful scenario doesn't affect. Success scenario is the most commonly used flow. Is there anything that we can't set the flag earlier?

Reply:
I'm concerned that we may run into cases the flag is visible before we commit. However, I think unflagging the updates is alright currently as only this function uses this flag and eviction and checkpoint cannot run on the same page concurrently.

Reply:
I still prefer flagging the updates after commit. The code is a little bit nasty if we have to unflag the updates and I think the overhead is acceptable as we only flag the first update inserted to las.

Reply:
I tend to agree with @kommiharibabu but also I still think you should flag all the updates. It may be worth asking someone else to decide?